specs/release-automation/release-automation-specs.qmd

```diff
@@ -0,0 +1,230 @@
+# Release Automation
+
+This document will go over the specifications for a software system to automate the release process for the Ottawa Data Model (ODM).
+
+## Audience
+
+The primary audience for this document are software engineers who will be responsible for developing the system.
+
+## Context
+
+Wastewater surveillance enables public health departments to monitor communities for possible outbreaks of different infectious diseases using wastewater samples, most notably the different variants of the COVID-19 virus. The ODM dictionary is an open source data model used to represent wastewater surveillance data with all its documentation available [online](https://github.com/Big-Life-Lab/PHES-ODM).
+
+Practically, the dictionary is implemented as an Excel document. Although the main purpose of the excel sheet is to document the data model details in a machine actionable way, it also contains other sheets, for example data templates that make it easy for users to input their wastewater data.
+
+Releasing a new version of the dictionary is a laborious process that requires converting the excel document to multiple output formats. In addition, the different release files need to be uploaded to multiple release locations. Details about the release process are available [online](https://odm.discourse.group/t/generation-of-tables-and-lists-from-the-odm-working-excel-file/99/7). The current manual process of implementing a release takes time away from the dictionary developers as well as is susceptible to errors. Automating this process would increase the quality of each release as well as give back time to the dictionary staff.
+
+## Software Constraints
+
+* The software system will need to use GitHub actions as its continuous integration tool.
+* The software system will need to be written in R or Python.
+
+## Features
+
+### RA-1: Trigerring the Process
+
+A user will need to manually triger the release process from the [Github Actions tab](https://github.com/Big-Life-Lab/PHES-ODM/actions) in the [PHES-ODM repo](https://github.com/Big-Life-Lab/PHES-ODM). The following inputs will need to be provided by the user:
+
+1. Link to the excel dictionary to use for the release. Currently, only links to an OSF repo are allowed.
+2. The OSF personal access token to use. The system will need this to gain access to the repo and perform operations on it.
+
+### RA-2: Creating the Release Files
```

Comment:
I couldn't follow this section. I can't find all the headings, which may be just me. I think it best for you to walk me through this section while we look at the tab together.